test/integration/WovnIndexSampleApacheTest.php

```diff
@@ -0,0 +1,225 @@
+<html ⚡><head></head><body>test</body></html>
+EXPECTED;
+
+        $this->writeFile('amp.php', $amp_php_content);
+        $this->setWovnIni($this->getWovnIni(array('check_amp' => 1)));
+
+        $response = $this->fetchURL('/amp.php');
+
+        $this->assertEquals($expected, $response->body);
+    }
+
+    public function testStaticHtml()
+    {
+        $static_html_content = <<<CONTENT
+<html>
+  <head></head>
+  <body>
+    <h1>Static Content</h1>
+  </body>
+</html>
+CONTENT;
+
+        $expected = <<<EXPECTED
+<html>
+  <head><link rel="alternate" hreflang="ja" href="http://localhost/static.html?a=b&amp;wovn=ja"><link rel="alternate" hreflang="fr" href="http://localhost/static.html?a=b&amp;wovn=fr"><link rel="alternate" hreflang="bg" href="http://localhost/static.html?a=b&amp;wovn=bg"><link rel="alternate" hreflang="en" href="http://localhost/static.html?a=b"><script src="//j.wovn.io/1" data-wovnio="key=Tek3n&amp;backend=true&amp;currentLang=en&amp;defaultLang=en&amp;urlPattern=query&amp;langCodeAliases=[]&amp;langParamName=wovn" data-wovnio-info="version=WOVN.php_VERSION" async></script></head>
+  <body>
+    <h1>Static Content</h1>
+  </body>
+</html>
+EXPECTED;
+
+        $this->writeFile('static.html', $static_html_content);
+        $this->setWovnIni($this->getWovnIni());
+
+        $response = $this->fetchURL('/static.html?a=b');
+
+        $this->assertEquals($expected, $response->body);
+    }
+
+    private function fetchURL($path)
+    {
+        return Utils::fetchURL('http://localhost' . $path);
+    }
+
+    private function setMockApiResponse($contents)
+    {
+        $this->writeFile('v0/translation', $contents);
+    }
+
+    private function setWovnIni($contents)
+    {
+        $this->writeFile('wovn.ini', $contents);
+    }
+
+    private function setHtaccess($contents)
+    {
+        $this->writeFile('.htaccess', $contents);
+    }
+
+    private function writeFile($file, $contents)
+    {
+        $filePath = $this->docRoot . '/' . $file;
+
+        $content = is_array($contents) ? implode("\n", $contents) : $contents;
+        file_put_contents($filePath, $content);
+    }
+
+    private function getWovnIni($options = array())
+    {
+        $defaultOptions = array(
+            'project_token' => 'Tek3n',
+            'url_pattern_name' => 'query',
+            'default_lang' => 'en',
+            'encoding' => 'UTF-8',
+            'disable_api_request_for_default_lang' => 'true',
+            'supported_langs' => array('ja', 'fr', 'bg', 'en'),
+            'api_url' => 'http://localhost/v0/',
+        );
+        $options = array_merge($defaultOptions, $options);
+
+        $contents = array();
+        foreach ($options as $name => $value) {
+            if (is_array($value)) {
+                foreach ($value as $v) {
+                    $contents[] = "${name}[] = $v";
+                }
+            } else {
+                $contents[] = "$name = $value";
+            }
+        }
```

Comment:
@Soichiro725 
> I think this method doesn't work for custom lang code.

So custom lang code is a problem because it's structured like this, right?
`['custom_lang_aliases'] = array('ja' => ' Japanese');`
I am going to change the code as follows, what do you think?
```suggestion
        foreach ($options as $name => $value) {
            if (is_array($value)) {
                foreach ($value as $key => $v) {
                    if (is_string($key)) {
                        $contents[] = "${name}[$key] = $v";
                    } else {
                        $contents[] = "${name}[] = $v";
                    }
                }
            } else {
                $contents[] = "$name = $value";
            }
        }
```

Reply:
I think converting from setting to ini file cause complexity.
How about remove `getWovnIni` method and create wovn.ini via `setWovnIni`.
```
private function setHtaccess($options)
{
  $defaultOptions = array(
    'project_token => "Tek3n"',
    'api_url = "http://localhost/v0/"'
  );
  $merged_options = array_merge($defaultOptions, $options);
  $wovn_ini = implode("\n", $merged_options);
  $this->writeFile('wovn.ini', $wovn_ini);
}
```

Reply:
Is this solved?